src/packageurl/contrib/django_models.py

```diff
@@ -120,7 +122,7 @@ def set_package_url(self, package_url):
         if not isinstance(package_url, PackageURL):
             package_url = PackageURL.from_string(package_url)
 
-        for field_name, value in package_url.to_dict(encode=True).items():
+        for field_name, value in package_url.to_dict().items():
```

Comment:
Can you elaborate a bit on that? And also for these changes, we might want to have a small test? or may this is not needed?

Reply:
Here we go : 

Setting `encode=True` we get this : 

```
In [1]: from packageurl import PackageURL                                                                                                                                           

In [2]: p1 = PackageURL(name="cream",type="ice",version="1.0",qualifiers={"flavour":"mango"})                                                                                       

In [3]: from vulnerabilities.models import Package                                                                                                                                  

In [4]: a = Package()                                                                                                                                                               

In [5]: a.set_package_url(p1)                                                                                                                                                       

In [6]: a                                                                                                                                                                           
Out[6]: <Package: pkg:ice/cream@1.0?flavour=mango>

In [7]: a.qualifiers                                                                                                                                                                
Out[7]: 'flavour=mango'
```

After removing `encode=True` in the `to_dict` method 
```
In [1]: from packageurl import PackageURL                                                                                                                                           

In [2]:  p1 = PackageURL(name="cream",type="ice",version="1.0",qualifiers={"flavour":"mango"})                                                                                                                                                                                                                                                                 

In [3]: from vulnerabilities.models import Package                                                                                                                                                                                                                                                                                            

In [4]: a = Package()                                                                                                                                                               

In [5]: a.set_package_url(p1)                                                                                                                                                       

In [6]: a.qualifiers                                                                                                                                                                
Out[6]: OrderedDict([('flavour', 'mango')])

```

As you can see setting `encode=True`  normalizes and converts the qualifiers into a string. So even if we use JSONField, that would still save qualifers as a string(rather than key-value in the qualifers db column)

**Note** : `PackageURLMixin` and `vulnerabilties.models.Package` are basically the same thing  with the difference `Package` having a  field for FK of `Vulnerabilty`